x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -1184,9 +1184,16 @@ export class Embeddable
     }
 
     const hasData = Object.values(this.activeData).some((table) => {
-      const count = table.meta?.statistics?.totalCount || table.rows.length;
-
-      return count > 0;
+      if (
+        table.meta &&
+        table.meta.statistics &&
+        typeof table.meta.statistics.totalCount === 'number'
+      ) {
```

Comment:
I see from its type declaration that `....totalCount?: number | undefined`.
Filtering out the `undefined` case should be enough here I think:

```suggestion
      if (table.meta?.statistics?.totalCount != null) {
```